lib/ansible/modules/packaging/os/zypper.py

```diff
@@ -57,9 +57,12 @@
           - C(present) will make sure the package is installed.
             C(latest)  will make sure the latest version of the package is installed.
             C(absent)  will make sure the specified package is not installed.
+            C(dist-upgrade) will make sure the latest version of all installed packages from all enabled repositories is installed.
+          - When using C(dist-upgrade), I(name) should be C('*').
         required: false
-        choices: [ present, latest, absent ]
+        choices: [ present, latest, absent, dist-upgrade ]
         default: "present"
+        version_added: "2.3"
```

Comment:
The `version_added` is not correct here, as it refers to `state` and not `dist-upgrade`. One could refer to the version in the doc string above, but I'd leave it out since it's an addition and the error message in old versions is descriptive enough.